Cython/Compiler/ExprNodes.py

```diff
@@ -15268,7 +15269,9 @@ def analyse_types(self, env):
                 # This is a special case of https://github.com/cython/cython/issues/4146
                 # TODO - once that's fixed general revisit this code and possibly
                 # use coerce_to_simple
-                self.assignment.rhs = copy.copy(self.rhs)
+                self.assignment.rhs = copy.copy(self.rhs.arg)
+                self.rhs = self.rhs.arg  # take out of ProxyNode
+                self.is_literal_assignment = True
```

Comment:
I find this ordering a little clearer.
```suggestion
                self.rhs = self.rhs.arg  # take out of ProxyNode
                self.assignment.rhs = copy.copy(self.rhs)  # replace CloneNode with copy
                self.is_literal_assignment = True
```
Also, something like `self.assignment_is_independent` or "_separate" might catch the intention a bit better.